### ISSUE TITLE
In Tensorflow Lite, Instance norm takes a lot of time

### ISSUE THREAD

**Issue by czczup** [tensorflow/tensorflow#38397]
**System information**
- OS Platform and Distribution: window10
- TensorFlow installed from (source or binary):binary
- TensorFlow version (or github SHA if from source):1.15.0

**Provide the text output from tflite_convert**
```
2020-04-09 23:44:06.327454: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2
2020-04-09 23:44:06.620150: I tensorflow/core/grappler/devices.cc:60] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0 (Note: TensorFlow was not compiled with CUDA support)
2020-04-09 23:44:06.631412: I tensorflow/core/grappler/clusters/single_machine.cc:356] Starting new session
2020-04-09 23:44:06.870610: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:786] Optimization results for grappler item: graph_to_optimize
2020-04-09 23:44:06.879972: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:788]   constant_folding: Graph size after: 295 nodes (-88), 348 edges (-90), time = 80.397ms.
2020-04-09 23:44:06.889699: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:788]   constant_folding: Graph size after: 295 nodes (0), 348 edges (0), time = 26.95ms.
```

**Standalone code to reproduce the issue** 
I'm trying to run a [Fast Style Transfer](https://github.com/lengstrom/fast-style-transfer) model in my Android phone. I find when I use batchnorms in this network, it takes very little time (40ms), but when I use instance norms, it takes a lot of time (215ms). 

The Batch norm mentioned here is `tf.layers.batch_normalization()`. The CPU of my Android phone is snapdragon 845, and in TensorFlow Lite I use the GPU mode (GpuDelegate). 

[Image Transform network](https://github.com/lengstrom/fast-style-transfer/blob/master/src/transform.py):
```
def net(image):
    conv1 = _conv_layer(image, 32, 9, 1)
    conv2 = _conv_layer(conv1, 64, 3, 2)
    conv3 = _conv_layer(conv2, 128, 3, 2)
    resid1 = _residual_block(conv3, 3)
    resid2 = _residual_block(resid1, 3)
    resid3 = _residual_block(resid2, 3)
    resid4 = _residual_block(resid3, 3)
    resid5 = _residual_block(resid4, 3)
    conv_t1 = _conv_tranpose_layer(resid5, 64, 3, 2)
    conv_t2 = _conv_tranpose_layer(conv_t1, 32, 3, 2)
    conv_t3 = _conv_layer(conv_t2, 3, 9, 1, relu=False)
    preds = tf.add(tf.nn.tanh(conv_t3) * 150,255./2, name="output")
    return preds
```
Instance norm:
```
def _instance_norm(net, train=True):
    batch, rows, cols, channels = [i.value for i in net.get_shape()]
    var_shape = [channels]
    mu, sigma_sq = tf.nn.moments(net, [1, 2], keep_dims=True)
    shift = tf.Variable(tf.zeros(var_shape))
    scale = tf.Variable(tf.ones(var_shape))
    epsilon = 1e-3
    normalized = (net-mu)*tf.rsqrt((sigma_sq + epsilon))
    return scale * normalized + shift
```
I used the following version of TensorFlow Lite:
```
implementation 'org.tensorflow:tensorflow-lite:0.0.0-nightly'
implementation 'org.tensorflow:tensorflow-lite-gpu:0.0.0-nightly'
implementation 'org.tensorflow:tensorflow-lite-support:0.0.0-nightly'
```

**Any other info / logs**
When I use the Instance norm, the model requires an average of 215ms in CPU mode and 205ms in GPU mode (stylize a image of 128*128 pixels). I'm sure the GPU is working, but it seems like the Instance norm is running on the CPU, so there's no obvious time decrease. How can I improve the speed of the instance norm in TensorFlow Lite？

**Comment by czczup** [comment]
[model_batchnorm](https://drive.google.com/file/d/1lJQCnTCNypn57M31DtktRJ0YaBLAlUkP/view?usp=sharing)
[model_instance_norm](https://drive.google.com/file/d/1TOdpWbOTTsVLVUMWROS_Sde5rKBXCP38/view?usp=sharing)
[model_without_norm](https://drive.google.com/file/d/1l4c0MGoEKN_m177tcodHdx_rNTzVYtIY/view?usp=sharing)

input shape: 1\*128\*128\*3, tf.float32
output shape: 1\*128\*128\*3, tf.float32 

my test results (num_thread=4):
| model | CPU | GPU |
|  :----: |  :----: | :----: |
| model_instance_norm | 215.7ms | 205.1ms |
| model_batch_norm | 169.6ms | 40.6ms |
| model_without_norm | 167.7ms | 41.4ms |

**Comment by abattery** [comment]
There are some unsupported ops of GPU delegate in the instance norm graph.
https://www.tensorflow.org/lite/performance/gpu_advanced#supported_ops

**Comment by czczup** [comment]
> There are some unsupported ops of GPU delegate in the instance norm graph.
> https://www.tensorflow.org/lite/performance/gpu_advanced#supported_ops

I investigated the difference between Batch norm and Instance norm, I found that the Batch norm layer will calculate the moving mean and moving variance when training, but use the learned moving value when testing, so the `tf.nn.moments` was only used at training time in BN layer. I think is the reason why Batch norm is faster than Instance norm. I'm not sure this idea is right.

**Comment by abattery** [comment]
In mobile environment, to use GPU acceleration fully, the ops in the graph should be supported by the TensorFlow Lite GPU delegate. The instance norm based TFLite graph contains some unsupported ops of GPU delegate. Like your performance experiments, the executions of many parts of the graph will be fall back to CPU calculations.

**Comment by czczup** [comment]
I solved this problem using other framework.

**Comment by terryheo** [comment]
That list of supported ops is bit outdated. You can find the actual list from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/common/model_builder.cc#L2447

I've tested the model_instance_norm and it works well with GPU delegate.
In Pixel 4, GPU takes 15ms while CPU takes 263ms.
It might be worth to try it with recent TFLite versions.

**Comment by czczup** [comment]
> That list of supported ops is bit outdated. You can find the actual list from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/common/model_builder.cc#L2447
> 
> I've tested the model_instance_norm and it works well with GPU delegate.
> In Pixel 4, GPU takes 15ms while CPU takes 263ms.
> It might be worth to try it with recent TFLite versions.

Which version do you use?

**Comment by terryheo** [comment]
> Which version do you use?

I used master branch.
